### ISSUE TITLE
Reproduce your reported numbers in paper

### ISSUE THREAD

**Issue by Lilyo** [dairui01/PDAN#2]
Hi, 

Thank you in advance for your excellent work!

Could you please provide guidance or something for evaluation? We wanna reproduce your reported numbers in paper.

In fact, the best eval mAP we got during training was 19.79...

I'm looking forward to your reply!

Thanks

**Comment by dairui01** [comment]
Hi, 

Thank you for your interest in this work.

The evaluation is the f-mAP, which is provided in apmeter.py. This metric is the same as the other STOA methods (TGM, superevent, MLAD). By using the 'average_precision_score' in sklearn.metrics, you could get something similar. 

If you can not get the correct result, firstly, you can change the initial learning rate to a relatively small value (e.g 2E-4) to check if the results go well. In this way, you don't need to wait for the scheduler to adjust the learning rate. 
For your convenience, here is the pre-trained model on RGB ([LINK](https://mybox.inria.fr/f/e8feeb538ec445a3ace1/?dl=1)). Please read the README file first. 
 

Also note that the input I3D feature is critical to the model performance. Please make sure that you are following AJ's Pytorch-I3D using their pre-trained model to extract the feature (the image resolution is also important). The difference between the results may be caused by this input I3D feats as well.

**Comment by Lilyo** [comment]
Hi @dairui01 , 
Thank you for your reply! I firmly sure that I3D feature was well extracted because I have also applied to other work and it works well.

After you updated some files and solved the missing parameters in the previous submission, I got better performance. However, it still cannot reach 23.8 mAP. So I provide logs following for you reference. Hope you can help check the code again. 

Thanks!

```
Random_SEED!!!: 0
PDAN
batch_size: 16
cuda_avail True
RGB mode ./../Datasets/npy
load data ./../Datasets/npy
split!!!! training
100%|███████████████████████████████████████████████████████████████████████████████
██████████████████████████████████████████████████████████████████████| 9848/9848 [0
0:02<00:00, 3443.16it/s]
split!!!! testing
100%|███████████████████████████████████████████████████████████████████████████████
█████████████████████████████████████████████████████████████████████| 9848/9848 [00
:00<00:00, 13470.73it/s]
you are processing PDAN
pytorch_total_params 5850781
num_channel: 512 input_channnel: 1024 num_classes: 157
0.0001
Epoch 0/99
----------
train-map: tensor(5.7416)
val-map: tensor(14.4072)
Epoch 1/99
----------
train-map: tensor(19.1583)
val-map: tensor(18.1841)
Epoch 2/99
----------
train-map: tensor(23.6664)
val-map: tensor(19.5196)
Epoch 3/99
----------
train-map: tensor(28.0598)
val-map: tensor(20.2903)
Epoch 4/99
----------
train-map: tensor(29.6220)
val-map: tensor(21.0310)
Epoch 5/99
----------
train-map: tensor(31.5601)
val-map: tensor(21.3613)
Epoch 6/99
----------
train-map: tensor(34.1921)
val-map: tensor(21.5840)
Epoch 7/99
----------
train-map: tensor(34.0325)
val-map: tensor(21.7894)
Epoch 8/99
----------
train-map: tensor(34.5227)
val-map: tensor(21.8381)
Epoch 9/99
----------
train-map: tensor(36.3576)
val-map: tensor(22.0705)
Epoch 10/99
----------
train-map: tensor(36.3568)
val-map: tensor(22.0325)
Epoch 11/99
----------
train-map: tensor(40.8987)
val-map: tensor(22.2826)
Epoch 12/99
----------
train-map: tensor(41.6441)
val-map: tensor(22.1545)
Epoch 13/99
----------
train-map: tensor(42.3792)
val-map: tensor(22.1999)
Epoch 14/99
----------
train-map: tensor(42.9744)
val-map: tensor(22.1332)
Epoch 15/99
----------
train-map: tensor(45.4047)
val-map: tensor(22.0928)
Epoch    16: reducing learning rate of group 0 to 5.0000e-05.
Epoch 16/99
----------
train-map: tensor(48.7658)
val-map: tensor(22.0376)
Epoch 17/99
----------
train-map: tensor(48.5278)
val-map: tensor(22.0898)
Epoch 18/99
----------
train-map: tensor(49.8607)
val-map: tensor(22.0768)
Epoch 19/99
----------
train-map: tensor(49.2570)
val-map: tensor(21.9862)
Epoch 20/99
----------
train-map: tensor(52.3927)
val-map: tensor(21.9380)
Epoch 21/99
----------
train-map: tensor(53.3316)
val-map: tensor(21.9099)
Epoch 22/99
----------
train-map: tensor(52.3118)
val-map: tensor(21.8074)
Epoch 23/99
----------
train-map: tensor(52.6371)
val-map: tensor(21.7444)
Epoch 24/99
----------
train-map: tensor(52.7571)
val-map: tensor(21.5493)
Epoch    25: reducing learning rate of group 0 to 2.5000e-05.
Epoch 25/99
----------
train-map: tensor(56.0317)
val-map: tensor(21.6357)
Epoch 26/99
----------
train-map: tensor(56.2461)
val-map: tensor(21.5504)
Epoch 27/99
----------
train-map: tensor(56.9260)
val-map: tensor(21.4119)
Epoch 28/99
----------
train-map: tensor(56.4744)
val-map: tensor(21.4081)
Epoch 29/99
----------
train-map: tensor(56.3987)
val-map: tensor(21.3831)
Epoch 30/99
----------
train-map: tensor(57.0916)
val-map: tensor(21.3380)
Epoch 31/99
----------
train-map: tensor(58.4115)
val-map: tensor(21.3347)
Epoch 32/99
----------
train-map: tensor(57.6359)
val-map: tensor(21.3523)
Epoch 33/99
----------
train-map: tensor(60.1158)
val-map: tensor(21.2256)
Epoch    34: reducing learning rate of group 0 to 1.2500e-05.
Epoch 34/99
----------
train-map: tensor(58.2547)
val-map: tensor(21.1941)
Epoch 35/99
----------
train-map: tensor(59.9844)
val-map: tensor(21.1821)
Epoch 36/99
----------
train-map: tensor(59.1862)
val-map: tensor(21.1172)
Epoch 37/99
----------
train-map: tensor(58.3791)
val-map: tensor(21.0770)
Epoch 38/99
----------
train-map: tensor(57.2033)
val-map: tensor(21.0303)
Epoch 39/99
----------
train-map: tensor(59.7367)
val-map: tensor(21.0497)
Epoch 40/99
----------
train-map: tensor(61.7699)
val-map: tensor(21.0624)
Epoch 41/99
----------
train-map: tensor(62.3769)
val-map: tensor(21.0084)
Epoch 42/99
----------
train-map: tensor(58.7346)
val-map: tensor(20.9547)
Epoch    43: reducing learning rate of group 0 to 6.2500e-06.
Epoch 43/99
----------
train-map: tensor(61.3322)
val-map: tensor(20.9929)
Epoch 44/99
----------
train-map: tensor(63.4453)
val-map: tensor(20.9705)
Epoch 45/99
----------
train-map: tensor(61.7444)
val-map: tensor(20.9116)
Epoch 46/99
----------
train-map: tensor(60.9134)
val-map: tensor(20.9297)
Epoch 47/99
----------
train-map: tensor(60.6135)
val-map: tensor(20.8840)
Epoch 48/99
----------
train-map: tensor(62.8932)
val-map: tensor(20.8272)
Epoch 49/99
----------
train-map: tensor(61.8009)
val-map: tensor(20.8510)
Epoch 50/99
----------
train-map: tensor(63.1793)
val-map: tensor(20.8408)
Epoch 51/99
----------
train-map: tensor(66.1910)
val-map: tensor(20.8279)
Epoch    52: reducing learning rate of group 0 to 3.1250e-06.
Epoch 52/99
----------
train-map: tensor(62.2591)
val-map: tensor(20.8156)
Epoch 53/99
----------
train-map: tensor(64.7813)
val-map: tensor(20.8227)
Epoch 54/99
----------
train-map: tensor(63.1430)
val-map: tensor(20.7979)
Epoch 55/99
----------
train-map: tensor(62.9620)
val-map: tensor(20.8058)
Epoch 56/99
----------
train-map: tensor(60.8277)
val-map: tensor(20.7994)
Epoch 57/99
----------
train-map: tensor(61.0806)
val-map: tensor(20.7713)
Epoch 58/99
----------
train-map: tensor(63.2845)
val-map: tensor(20.7747)
Epoch 59/99
----------
train-map: tensor(61.1625)
val-map: tensor(20.7586)
Epoch 60/99
----------
train-map: tensor(62.5408)
val-map: tensor(20.7684)
Epoch    61: reducing learning rate of group 0 to 1.5625e-06.
Epoch 61/99
----------
train-map: tensor(62.2107)
val-map: tensor(20.7619)
Epoch 62/99
----------
train-map: tensor(63.1032)
val-map: tensor(20.7677)
Epoch 63/99
----------
train-map: tensor(62.3870)
val-map: tensor(20.7586)
Epoch 64/99
----------
train-map: tensor(62.0379)
val-map: tensor(20.7432)
Epoch 65/99
----------
train-map: tensor(62.6745)
val-map: tensor(20.7393)
Epoch 66/99
----------
train-map: tensor(62.0780)
val-map: tensor(20.7349)
Epoch 67/99
----------
train-map: tensor(63.2793)
val-map: tensor(20.7345)
Epoch 68/99
----------
train-map: tensor(61.3541)
val-map: tensor(20.7302)
Epoch 69/99
----------
train-map: tensor(62.4179)
val-map: tensor(20.7406)
Epoch    70: reducing learning rate of group 0 to 7.8125e-07.
Epoch 70/99
----------
train-map: tensor(64.4249)
val-map: tensor(20.7373)
Epoch 71/99
----------
train-map: tensor(65.0813)
val-map: tensor(20.7217)
Epoch 72/99
----------
train-map: tensor(64.4197)
val-map: tensor(20.7177)
Epoch 73/99
----------
train-map: tensor(64.3789)
val-map: tensor(20.7185)
Epoch 74/99
----------
train-map: tensor(63.0715)
val-map: tensor(20.7275)
Epoch 75/99
----------
train-map: tensor(65.0526)
val-map: tensor(20.7195)
Epoch 76/99
----------
train-map: tensor(64.3179)
val-map: tensor(20.7161)
Epoch 77/99
----------
train-map: tensor(62.0232)
val-map: tensor(20.7028)
Epoch 78/99
----------
train-map: tensor(63.9726)
val-map: tensor(20.6974)
Epoch    79: reducing learning rate of group 0 to 3.9063e-07.
Epoch 79/99
----------
train-map: tensor(64.5381)
val-map: tensor(20.7028)
Epoch 80/99
----------
train-map: tensor(63.2612)
val-map: tensor(20.7027)
Epoch 81/99
----------
train-map: tensor(63.4041)
val-map: tensor(20.7050)
Epoch 82/99
----------
train-map: tensor(62.9297)
val-map: tensor(20.7023)
Epoch 83/99
----------
train-map: tensor(63.1827)
val-map: tensor(20.6958)
Epoch 84/99
----------
train-map: tensor(63.6678)
val-map: tensor(20.6957)
Epoch 85/99
----------
train-map: tensor(63.6974)
val-map: tensor(20.6965)
Epoch 86/99
----------
train-map: tensor(61.7784)
val-map: tensor(20.6970)
Epoch 87/99
----------
train-map: tensor(64.4029)
val-map: tensor(20.6890)
Epoch    88: reducing learning rate of group 0 to 1.9531e-07.
Epoch 88/99
----------
train-map: tensor(63.6852)
val-map: tensor(20.6932)
Epoch 89/99
----------
train-map: tensor(63.8013)
val-map: tensor(20.6922)
Epoch 90/99
----------
train-map: tensor(60.6742)
val-map: tensor(20.6928)
Epoch 91/99
----------
train-map: tensor(63.6413)
val-map: tensor(20.6924)
Epoch 92/99
----------
train-map: tensor(63.4539)
val-map: tensor(20.6929)
Epoch 93/99
----------
train-map: tensor(65.1603)
val-map: tensor(20.6921)
Epoch 94/99
----------
train-map: tensor(63.0044)
val-map: tensor(20.6896)
Epoch 95/99
----------
train-map: tensor(62.4869)
val-map: tensor(20.6899)
Epoch 96/99
----------
train-map: tensor(63.5058)
val-map: tensor(20.6902)
Epoch    97: reducing learning rate of group 0 to 9.7656e-08.
Epoch 97/99
----------
train-map: tensor(63.1332)
val-map: tensor(20.6905)
Epoch 98/99
----------
train-map: tensor(63.5121)
val-map: tensor(20.6932)
Epoch 99/99
train-map: tensor(63.2325)
val-map: tensor(20.3452)

```
